pyaedt/modeler/cad/Primitives2D.py

```diff
@@ -182,6 +196,7 @@ def create_rectangle(self, position, dimension_list, is_covered=True, name=None,
         ...                                          name="MyCircle", matname="Copper")
 
         """
+        # TODO: Primitives in Maxwell 2D must have Z=0, otherwise the transparency cannot be changed. (issue 4071)
```

Comment:
@dcrawforAtAnsys regarding the transparency issue I'd put a check in _default_object_attributes because we set the transparency either 0.2 or 0.8 depending the material name, so here I'd add an additional check for the coordinates.